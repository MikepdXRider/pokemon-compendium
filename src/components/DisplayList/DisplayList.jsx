import React from 'react'
import DisplayCard from '../DisplayCard/DisplayCard.jsx'

export default function DisplayList({dataArr}) {
    return (
        <section>
            <ul>   
                {
                    dataArr.map(dataObj => <li key={dataObj.pokemon}> <DisplayCard dataObj={dataObj} /> </li>)
                }
            </ul>
        </section>
    )
}
