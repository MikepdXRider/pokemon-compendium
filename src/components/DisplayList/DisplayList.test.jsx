import { render } from "@testing-library/react";
import DisplayList from "./DisplayList.jsx";

const dataArr = [{
        pokemon: 'test-name',
        hp: 10,
        base_experience: 15,
        type_1: 'test-type'
    }]

it('matches snapshot', () => {
    const {container} = render(<DisplayList dataArr={dataArr}/> );
    expect(container).toMatchSnapshot();
})