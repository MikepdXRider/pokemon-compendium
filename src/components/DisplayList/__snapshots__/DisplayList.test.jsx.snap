// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <section>
    <ul>
      <li>
         
        <article>
          <h3>
            test-name
          </h3>
          <img
            alt="test-name"
          />
          <p>
            HP: 10
          </p>
          <p>
            Base Experience: 15
          </p>
          <p>
            Type: test-type
          </p>
        </article>
         
      </li>
    </ul>
  </section>
</div>
`;
